core/runtime/src/main/java/io/quarkus/runtime/configuration/FileSystemOnlySourcesConfigBuilder.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.runtime.configuration;
+
+import static io.smallrye.config.PropertiesConfigSourceLoader.inFileSystem;
+
+import java.nio.file.Paths;
+
+import io.smallrye.config.SmallRyeConfigBuilder;
+
+public class FileSystemOnlySourcesConfigBuilder implements ConfigBuilder {
+    @Override
+    public SmallRyeConfigBuilder configBuilder(final SmallRyeConfigBuilder builder) {
+        return builder.withSources(
+                inFileSystem(Paths.get(System.getProperty("user.dir"), "config", "application.properties").toUri().toString(),
+                        260, Thread.currentThread().getContextClassLoader()));
```

Comment:
Use the classloader set in the builder: `io.smallrye.config.SmallRyeConfigBuilder#getClassLoader`.